### ISSUE TITLE
CC64 midi controller not working

### ISSUE THREAD

**Issue by Exink** [greatest-ape/OctaSine#119]
Hi, I'd like to suggest that this plug-in could handle the MIDI CC64 hold controller (or any other foot /sustain controller)), so that we could use it for piano like patches or anything similar. I haven't being able to make it work, that's why I'm opening this issue.

If there's some way to use this feature, please let me know. Otherwise hopefully this could be implemented in future releases.

Thanks a lot for making this great synth!

Best regards.

**Comment by greatest-ape** [comment]
No such support is implemented at the moment. Is the concept that while the pedal is pressed down, notes stay on (stay in sustain phase) even if the corresponding keys are no longer pressed?

**Comment by Exink** [comment]
Yes, that's supposed to happen, so this concept is more similar to the sustain-pedal of a piano. I don't if other controllers work differently, but indeed once the controller 64 is activated, as lon as it's on, notes should keep playing even if yourelease it. But once is disabled and no more notes playing, there shouldn¡t be any other note playing.

**Comment by greatest-ape** [comment]
Could you please try out this build with support added, check if it works and see if there are any issues?

[OctaSine-tmp-2022-11-08-Windows.zip](https://github.com/greatest-ape/OctaSine/files/9965827/OctaSine-tmp-2022-11-08-Windows.zip)
[OctaSine-tmp-2022-11-08-macOS.zip](https://github.com/greatest-ape/OctaSine/files/9965842/OctaSine-tmp-2022-11-08-macOS.zip)
[OctaSine-tmp-2022-11-08-Ubuntu-20_04.zip](https://github.com/greatest-ape/OctaSine/files/9965828/OctaSine-tmp-2022-11-08-Ubuntu-20_04.zip)

**Comment by Exink** [comment]
That's perfect! I've just tested it on my personal DX7-EP1 patch recreation and it works as intended. Personally I haven't found any issue so far, even after playing various notes at once. Do you need an audio sample?

**Comment by greatest-ape** [comment]
Thanks, great! No, it’s ok :-)

I’ve merged the fix and it will be included in next release.

**Comment by Exink** [comment]
Ok, thanks a lot! Looking forward to the next one :).
